sagemaker-python-sdk/spot_training_tensorflow_estimator/spot_training_tensorflow_estimator.ipynb

```diff
@@ -0,0 +1,178 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install -qU awscli boto3 sagemaker"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "isConfigCell": true
+   },
+   "outputs": [],
+   "source": [
+    "from sagemaker import get_execution_role\n",
+    "from sagemaker.session import Session\n",
+    "\n",
+    "# S3 bucket for saving code and model artifacts.\n",
+    "# Feel free to specify a different bucket here if you wish.\n",
+    "bucket = Session().default_bucket()\n",
+    "\n",
+    "# Location to save your custom code in tar.gz format.\n",
+    "custom_code_upload_location = 's3://{}/customcode/tensorflow_iris'.format(bucket)\n",
+    "\n",
+    "# Location where results of model training are saved.\n",
+    "model_artifacts_location = 's3://{}/artifacts'.format(bucket)\n",
+    "\n",
+    "#IAM execution role that gives SageMaker access to resources in your AWS account.\n",
+    "role = get_execution_role()\n",
+    "\n",
+    "def estimator(model_path, hyperparameters):\n",
+    "    feature_columns = [tf.feature_column.numeric_column(INPUT_TENSOR_NAME, shape=[4])]\n",
+    "    return tf.estimator.DNNClassifier(feature_columns=feature_columns,\n",
+    "                                      hidden_units=[10, 20, 10],\n",
+    "                                      n_classes=3,\n",
+    "                                      model_dir=model_path)\n",
+    "\n",
+    "def estimator(model_path, hyperparameters):\n",
+    "    feature_columns = [tf.feature_column.numeric_column(INPUT_TENSOR_NAME, shape=[4])]\n",
+    "    return tf.estimator.DNNClassifier(feature_columns=feature_columns,\n",
+    "                                      hidden_units=[10, 20, 10],\n",
+    "                                      n_classes=3,\n",
+    "                                      model_dir=model_path)\n",
+    "\n",
+    "def train_input_fn(training_dir, hyperparameters):\n",
+    "    training_set = tf.contrib.learn.datasets.base.load_csv_with_header(\n",
+    "        filename=os.path.join(training_dir, 'iris_training.csv'),\n",
+    "        target_dtype=np.int,\n",
+    "        features_dtype=np.float32)\n",
+    "\n",
+    "    return tf.estimator.inputs.numpy_input_fn(\n",
+    "        x={INPUT_TENSOR_NAME: np.array(training_set.data)},\n",
+    "        y=np.array(training_set.target),\n",
+    "        num_epochs=None,\n",
+    "        shuffle=True)()\n",
+    "\n",
+    "def serving_input_fn(hyperparameters):\n",
+    "    feature_spec = {INPUT_TENSOR_NAME: tf.FixedLenFeature(dtype=tf.float32, shape=[4])}\n",
+    "    return tf.estimator.export.build_parsing_serving_input_receiver_fn(feature_spec)()\n",
+    "\n",
+    "import boto3\n",
```

Comment:
Minor: move this down to the cell where region gets used

Reply:
see answer to previous comment